python/dalex/dalex/arena/plots/_roc_container.py

```diff
@@ -14,29 +13,35 @@ class ROCContainer(PlotContainer):
     options = {
         'grid_points': { 'default': 101, 'desc': 'Maximum number of points for ROC curve' },
     }
+
     def _fit(self, model):
         exp = model.explainer
         if exp.model_type != 'classification':
             self.set_message('ROC plot is only available for classificators')
             return
 
         y_hat = exp.predict(exp.data) if exp.y_hat is None else exp.y_hat
-        df = pd.DataFrame({ 'y': exp.y.astype(bool), 'y_hat': y_hat }).sort_values('y_hat', ascending=False)
-        P_n = np.sum(df.y)
+        df = pd.DataFrame({ 'y': exp.y.astype(bool), 'y_hat': y_hat }).groupby('y_hat').sum().reset_index().sort_values('y_hat', ascending=False)
+
+        P_n = df.y.sum()
         N_n = df.shape[0] - P_n
         if P_n == 0 or N_n == 0:
             self.set_message('Provided dataset contains only positive or only negative cases.', 'error')
             return
+
+        df['TPR'] = df.y.cumsum() / P_n
+        df['TNR'] = 1 - ((1 - df.y).cumsum() / N_n)
+
         grid_points = self.get_option('grid_points')
-        df['TPR'] = np.cumsum(df.y) / P_n
-        df['TNR'] = 1 - (np.cumsum(1 - df.y) / N_n)
         if df.shape[0] > grid_points:
             df = df.sample(grid_points).sort_values('y_hat', ascending=False)
+
         self.data = {
             'cutoff': df['y_hat'].tolist(),
-            'specifity': df['TNR'].tolist(),
-            'sensivity': df['TPR'].tolist()
+            'specifity': [0] + df['TNR'].tolist(),
```

Comment:
I believe, that specificity should start with 1 instead of 0.

Reply:
fixed